src/main/kotlin/org/koitharu/kotatsu/parsers/site/all/HitomiLaParser.kt

```diff
@@ -656,12 +658,12 @@ internal class HitomiLaParser(context: MangaLoaderContext) : LegacyMangaParser(c
 			val hash = image.getString("hash")
 			val commonId = commonImageId()
 			val imageId = imageIdFromHash(hash)
-			val subDomain = 'a' + subdomainOffset(imageId)
+			val subDomain = subdomainOffset(imageId) + 1
 
 			MangaPage(
 				id = generateUid(hash),
-				url = "https://${getDomain("${subDomain}a")}/webp/$commonId$imageId/$hash.webp",
-				preview = "https://${getDomain("${subDomain}tn")}/webpsmalltn/${thumbPathFromHash(hash)}/$hash.webp",
+				url = "https://a${subDomain}.$cdnDomain/$commonId$imageId/$hash.avif",
+				preview = null, // need fix / alternative ?
```

Comment:
```suggestion
				preview = "https://${thumbSubdomain}tn.$cdnDomain/avifsmalltn/${thumbPathFromHash(hash)}/$hash.avif"
```

```kt
val thumbSubdomain = 'a' + subdomainOffset(imageId)
```

haven't tested it, but should work

Reply:
I will try it

Reply:
@AwkwardPeak7 it seems the preview images cannot be loaded because they dont exist, always show red cover...

Reply:
![{4E36E69C-61C4-4C72-A993-0F37A308B360}](https://github.com/user-attachments/assets/2146cd67-4608-44c0-9e16-e67925cc51b7)
¯\\\_(ツ)\_/¯

maybe kotatsu doesn't support avif?

try webp:
```"https://${thumbSubdomain}tn.$cdnDomain/webpsmalltn/${thumbPathFromHash(hash)}/$hash.webp"```

Reply:
> ![{4E36E69C-61C4-4C72-A993-0F37A308B360}](https://private-user-images.githubusercontent.com/48650614/428505503-2146cd67-4608-44c0-9e16-e67925cc51b7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM0MDU1MTAsIm5iZiI6MTc0MzQwNTIxMCwicGF0aCI6Ii80ODY1MDYxNC80Mjg1MDU1MDMtMjE0NmNkNjctNDYwOC00NGMwLTllMTYtZTY3OTI1Y2M1MWI3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzMxVDA3MTMzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZhZTNmZGEyMTliZTBlNTI2NWU4NjhhMjZkOTYzZDFiZmE5ZGI2MTMzNDY3ODk0MDZhODU5OTAxODZiYTk0MjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.7Ga13kQDzaP27lldTY_zuriStAEjMJPmv-gxFjMRy_0) ¯\_(ツ)_/¯
> 
> maybe kotatsu doesn't support avif?
> 
> try webp: `"https://${thumbSubdomain}tn.$cdnDomain/webpsmalltn/${thumbPathFromHash(hash)}/$hash.webp"`

kotatsu support avif format, but idk why it didnt load ¯\_(ツ)_/¯

Reply:
> ![{4E36E69C-61C4-4C72-A993-0F37A308B360}](https://private-user-images.githubusercontent.com/48650614/428505503-2146cd67-4608-44c0-9e16-e67925cc51b7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM0MDkxNDQsIm5iZiI6MTc0MzQwODg0NCwicGF0aCI6Ii80ODY1MDYxNC80Mjg1MDU1MDMtMjE0NmNkNjctNDYwOC00NGMwLTllMTYtZTY3OTI1Y2M1MWI3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzMxVDA4MTQwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUxOTg2ODMzYWM0ZmE1OTBlN2NjZjlhNTkxNDY4YjY1NGI0OGNjODZmMmE2NWY2OTJiMzJjNDk3ODE5MTEzNjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3OVZHm7yNa_zViTXxzr_WPqEIMoOun0djBuSlIpmJcQ) ¯\_(ツ)_/¯
> 
> maybe kotatsu doesn't support avif?
> 
> try webp: `"https://${thumbSubdomain}tn.$cdnDomain/webpsmalltn/${thumbPathFromHash(hash)}/$hash.webp"`

Solved with webp format :ok_hand: tks